templates/single-product/product-image.php

```diff
@@ -26,11 +26,17 @@
 
 $columns           = apply_filters( 'woocommerce_product_thumbnails_columns', 4 );
 $post_thumbnail_id = $product->get_image_id();
+if ( ! $post_thumbnail_id ) {
```

Comment:
We could move this logic to `woocommerce_show_product_images()`, so we avoid increasing the number of conditional code inside templates.

Reply:
@claudiosanches Thanks for the suggestion. Done in `84d6d6c`.